CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+# Top level CMakeLists.txt: setting up top level project dependencies and options
+cmake_minimum_required(VERSION 3.20)
+project(FIMS
+  CXX
+) # CXX is the language name
+
+# GoogleTest requires at least C++11
+set(CMAKE_CXX_STANDARD 11)
+
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  URL https://github.com/google/googletest/archive/refs/tags/release-1.11.0.zip
```

Comment:
would we need to change this from time to time? I wonder if there is a way to just use the latest release? Or maybe fixing to a particular release is intentional?

Reply:
We will change it, but not very often. We will stick with particular releases when using CMake, Google Test, and Google Benchmark. When there is a need to use newer versions of dependencies, we will conduct regression testing before switching to new versions.